lib/miquido_lints.yaml

```diff
@@ -0,0 +1,336 @@
+
+  language:
+    strict-raw-types: true
+    strict-casts: true
+    strict-inference: true
+
+  errors:
+    missing_required_param: error
+    missing_return: error
+    unawaited_futures: warning
+    unused_import: warning
+    prefer_const_constructors: info
+    prefer_const_constructors_in_immutables: info
+
+linter:
+  rules:
+    - always_declare_return_types
+    - always_put_control_body_on_new_line
+    - always_put_required_named_parameters_first
+    - always_require_non_null_named_parameters
+    # - always_specify_types - Incompatible with avoid_types_on_closure_parameters, which we prefer
+    - always_use_package_imports
+    - annotate_overrides
+    # - avoid_annotating_with_dynamic - Super problematic to fix with avoid_types_on_closure_parameters enabled
+    - avoid_bool_literals_in_conditional_expressions
+    - avoid_catches_without_on_clauses
+    - avoid_catching_errors
+    # - avoid_classes_with_only_static_members - We know, but somewhat this doesnt fell right and clean
+    - avoid_double_and_int_checks
+    - avoid_dynamic_calls
+    - avoid_empty_else
+    - avoid_equals_and_hash_code_on_mutable_classes
+    - avoid_escaping_inner_quotes
+    - avoid_field_initializers_in_const_classes
+    - avoid_final_parameters
+    - avoid_function_literals_in_foreach_calls
+    - avoid_implementing_value_types
+    - avoid_init_to_null
+    - avoid_js_rounded_ints
+    - avoid_multiple_declarations_per_line
+    - avoid_null_checks_in_equality_operators
+    - avoid_positional_boolean_parameters
+    - avoid_print
+    - avoid_private_typedef_functions
+    - avoid_redundant_argument_values
+    - avoid_relative_lib_imports
+    - avoid_renaming_method_parameters
+    - avoid_return_types_on_setters
+    - avoid_returning_null
+    - avoid_returning_null_for_future
+    - avoid_returning_null_for_void
+    - avoid_returning_this
+    - avoid_setters_without_getters
+    - avoid_shadowing_type_parameters
+    - avoid_single_cascade_in_expression_statements
+    - avoid_slow_async_io
+    - avoid_type_to_string
+    - avoid_types_as_parameter_names
+    - avoid_types_on_closure_parameters
+    - avoid_unnecessary_containers
+    - avoid_unused_constructor_parameters
+    - avoid_void_async
+    - avoid_web_libraries_in_flutter
+    - await_only_futures
+    - camel_case_extensions
+    - camel_case_types
+    - cancel_subscriptions
+    - cascade_invocations
+    - cast_nullable_to_non_nullable
+    - close_sinks
+    - comment_references
+    - conditional_uri_does_not_exist
+    - constant_identifier_names
+    - control_flow_in_finally
+    - curly_braces_in_flow_control_structures
+    # - depend_on_referenced_packages - This will make you at some point add all pub.dev to your pubspec file. Maybe suitable for ultra small petprojets with one dependency at max.
+    - deprecated_consistency
+    # - diagnostic_describe_all_properties - Zero benefits found, add unncessary complexity to code
```

Comment:
well, maybe soften wording here? 
I think it's fairly useful in debugging and if we were developing a library, then this is something we probably would like to turn on, but in usual projects I agree that we don't want to spend time on it :)

Reply:
softened like a feather